sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregatesSuite.scala

```diff
@@ -30,7 +30,8 @@ class RemoveRedundantAggregatesSuite extends PlanTest {
 
   object Optimize extends RuleExecutor[LogicalPlan] {
     val batches = Batch("RemoveRedundantAggregates", FixedPoint(10),
-      RemoveRedundantAggregates) :: Nil
+      RemoveRedundantAggregates,
+      RemoveNoopOperators) :: Nil
```

Comment:
can we also add `CollpaseProject`? So that there is no adjacent `.select`

Reply:
`CollpaseProject` can not Collpase projects because `rand(0)` is non-deterministic. But there is still a little problem in `RemoveRedundantAggregatesSuite`, I will deal with it right away.

Reply:
The `rand(0)` is only referenced once, I think `CollpaseProject` can handle it?

Reply:
```
producer.deterministic && (count == 1 || alwaysInline || cheapToInlineProducer)
``` 
count is 1, but producer.deterministic is false.